### PR TITLE
IVR move month validation

### PR DESCRIPTION
**Pull request by NicolaasDieBaas**
move month validation for states 5A/B and 12A/B to a function and provide comments

### CODE REVIEW

**Comment by NicolaasDieBaas** [comment]
good testing for new function to be added at later stage

**Comment by NicolaasDieBaas** [comment]
@gsvr please review

File: src/utils.js [hunk 1]
```diff
@@ -711,5 +711,44 @@ go.utils = {
         return choices;
     },
 
+    // function used to validate months for states 5A/5B
+    is_valid_month_last_period: function(today, periodLastYear, choiceValue) {
+        if (periodLastYear) { today.subtract('year', 1); }     // substract year if last period happened the previous year
```
Comment (gsvr) [inline src/utils.js]: Even though there's no Pep8 for javascript that determines the max amount of characters per line, if you go over a certain amount like here it gets hard to review. So rather put the comment on the line above in this instance.

**Comment by NicolaasDieBaas** [comment]
@gsvr 2nd pass plz

**Comment by NicolaasDieBaas** [comment]
to implement the new make_month_choices utils function in all month choice states
&
refactor month validation logic

**Comment by NicolaasDieBaas** [comment]
@gsvr please have a look & let's discuss whether two reworked functions need to be merged into one

**Comment by NicolaasDieBaas** [comment]
@gsvr review plz

File: test/voice_registration.test.js [hunk 2]
```diff
@@ -1575,5 +1664,195 @@ describe("Mama Nigeria App", function() {
...
+                assert.equal(resultsForThisYearPeriod[11], false);    // dec
+
+                assert.equal(resultsForLastYearPeriod.length, 12);
+                assert.equal(resultsForLastYearPeriod[0], false);     // jan
+                assert.equal(resultsForLastYearPeriod[1], false);     // feb
+                assert.equal(resultsForLastYearPeriod[2], false);     // mar
+                assert.equal(resultsForLastYearPeriod[3], false);     // apr
+                assert.equal(resultsForLastYearPeriod[4], false);     // may
+                assert.equal(resultsForLastYearPeriod[5], false);     // jun
+                assert.equal(resultsForLastYearPeriod[6], true);      // jul
```
Comment (gsvr) [inline test/voice_registration.test.js]: `jul` should be false - this seems to be going back one month too far.

File: test/voice_registration.test.js [hunk 3]
```diff
@@ -1575,5 +1664,195 @@ describe("Mama Nigeria App", function() {
...
+                assert.equal(resultsForThisYearBaby[8], false);     // sep
+                assert.equal(resultsForThisYearBaby[9], false);     // oct
+                assert.equal(resultsForThisYearBaby[10], false);    // nov
+                assert.equal(resultsForThisYearBaby[11], false);    // dec
+
+                assert.equal(resultsForLastYearBaby.length, 12);
+                assert.equal(resultsForLastYearBaby[0], false);     // jan
+                assert.equal(resultsForLastYearBaby[1], false);     // feb
+                assert.equal(resultsForLastYearBaby[2], false);     // mar
+                assert.equal(resultsForLastYearBaby[3], true);      // apr
```
Comment (gsvr) [inline test/voice_registration.test.js]: `apr` should be false - this seems to be going back one month too far.

**Comment by gsvr** [comment]
:+1:

**Pull request merged**